### PR TITLE
Fix various svace issues

### PR DESCRIPTION
**Pull request by zhoonit**
- [svace] assure file to be closed before remove
```
 **Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```
- [svace] Error handling for applications/test
```
1. Fix inconsistent alloc/dealloc(new/free)
2. Add try catch to some statements
3. Fix memory leak from `asprintf`

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```
- [svace] Assure unintialized members
```
nnstreamer_layer had two unintialized members.
This patch initializes those two

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```

### CODE REVIEW

File: test/nntrainer_test_util.cpp [hunk 1]
```diff
@@ -136,6 +136,13 @@ int getBatch_train(float **outVec, float **outLabel, bool *last,
 
   if (!alloc_train) {
     duplicate = (bool *)malloc(sizeof(bool) * data_size);
+    if (duplicate == nullptr) {
+      ml_loge("[test_util] allocationg memory failed");
+      alloc_train = false;
+      *last = false;
```
Comment (kparichay) [inline test/nntrainer_test_util.cpp:142]: F.close()?

File: test/nntrainer_test_util.cpp [hunk 2]
```diff
@@ -213,6 +220,12 @@ int getBatch_val(float **outVec, float **outLabel, bool *last,
 
   if (!alloc_val) {
     valduplicate = (bool *)malloc(sizeof(bool) * data_size);
+    if (valduplicate == nullptr) {
+      ml_loge("[test_util] allocationg memory failed");
+      alloc_val = false;
+      *last = false;
+      return ML_ERROR_BAD_ADDRESS;
```
Comment (kparichay) [inline test/nntrainer_test_util.cpp:227]: ditto

**Review by kparichay — APPROVED** [review]
Overall LGTM other than the minor comment.

File: test/nntrainer_test_util.cpp (hunk 1 above)
Comment (zhoonit) [reply test/nntrainer_test_util.cpp:142]: Done

File: test/nntrainer_test_util.cpp (hunk 2 above)
Comment (zhoonit) [reply test/nntrainer_test_util.cpp:230]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**